tests/test_prompts.py

```diff
@@ -1,84 +1,80 @@
 import pytest
-from django.core.exceptions import ImproperlyConfigured
-from django.test import override_settings
-from wagtail_ai import DEFAULT_PROMPTS
-from wagtail_ai.prompts import get_prompt_by_id, get_prompts
+from django.core.exceptions import ValidationError
+from wagtail_ai.models import Prompt
+from wagtail_ai.prompts import DEFAULT_PROMPTS
+from wagtail_ai.wagtail_hooks import get_prompts
 
+# Apply pytest.mark.django_db to all tests in this file
+pytestmark = pytest.mark.django_db
 
-@override_settings(WAGTAIL_AI_PROMPTS=[])
-def test_get_prompts_returns_default_prompts():
-    prompts = get_prompts()
-    assert len(prompts) == len(DEFAULT_PROMPTS)
-    assert prompts[0].label == DEFAULT_PROMPTS[0]["label"]
-
-
-def test_prompts_are_given_ids():
-    prompts = get_prompts()
-    assert prompts[0].id is not None
-
-
-def test_get_prompt_by_id_returns_prompts_as_indexed():
-    assert get_prompt_by_id(0).label == DEFAULT_PROMPTS[0]["label"]
 
+def test_prompt_model(setup_prompt_object, test_prompt_values):
+    assert setup_prompt_object.is_default is False
+    assert str(setup_prompt_object) == test_prompt_values["label"]
+    assert setup_prompt_object.label == test_prompt_values["label"]
+    assert setup_prompt_object.prompt_value == test_prompt_values["prompt"]
+    assert setup_prompt_object.description == test_prompt_values["description"]
+    assert setup_prompt_object.method == test_prompt_values["method"]
 
-def test_custom_prompt(settings):
-    settings.WAGTAIL_AI_PROMPTS = [
-        {
-            "label": "Custom Prompt",
-            "description": "This is a custom prompt",
-            "prompt": "The prompt",
-            "method": "append",
-        }
-    ]
 
-    assert get_prompts()[0].label == "Custom Prompt"
+def test_get_prompts_returns_default_prompts():
+    prompts = get_prompts()
+    default_prompts_queryset = Prompt.objects.filter(
+        default_prompt_id__isnull=False
+    ).all()
 
+    assert len(default_prompts_queryset) == len(DEFAULT_PROMPTS)
```

Comment:
**Suggestion:** Testing length is not an indication of whether we get the correct result.

You should be testing what you get out of the query set, not the count or order. `assertCountEqual` is quite handy for that use case.

```python
from unittest import TestCase

TestCase().assertCountEqual(Prompt.objects.filter(
        default_prompt_id__isnull=False
    ).values_list("label", flat=True) , [p.label for p in DEFAULT_PROMPTS])
```